specification/agricultureplatform/AgriculturePlatform.DataPlane/routes.tsp

```diff
@@ -0,0 +1,512 @@
+  NoClientRequestId &
+  NoConditionalRequests;
+
+alias Operations = Azure.Core.ResourceOperations<ServiceTraits>;
+
+interface CatalogsOperations {
+  /**
+   * Returns a paginated list of Catalog resources.
+   */
+  list is Operations.ResourceList<
+    Catalog,
+    QueryParametersTrait<{
+      /**
+       * Names of the resource.
+       */
+      #suppress "@azure-tools/typespec-azure-core/no-query-explode" ""
+      @query(#{ name: "names", explode: true })
+      names?: string[];
+
+      /**
+       * Filters on key-value pairs within the Properties object.
+       * eg. "{testKey} eq {testValue}".
+       */
+      #suppress "@azure-tools/typespec-azure-core/no-query-explode" ""
+      @query(#{ name: "propertyFilters", explode: true })
+      propertyFilters?: string[];
+
+      /**
+       * Statuses of the resource.
+       */
+      #suppress "@azure-tools/typespec-azure-core/no-query-explode" ""
+      @query(#{ name: "statuses", explode: true })
+      statuses?: string[];
+
+      /**
+       * Ids of the resource.
+       */
+      #suppress "@azure-tools/typespec-azure-core/no-query-explode" ""
+      @query(#{ name: "ids", explode: true })
+      ids?: string[];
+
+      /**
+       * Minimum creation date of resource (inclusive).
+       */
+      @query("minCreatedDateTime")
+      minCreatedDateTime?: utcDateTime;
+
+      /**
+       * Maximum creation date of resource (inclusive).
+       */
+      @query("maxCreatedDateTime")
+      maxCreatedDateTime?: utcDateTime;
+
+      /**
+       * Minimum last modified date of resource (inclusive).
+       */
+      @query("minLastModifiedDateTime")
+      minLastModifiedDateTime?: utcDateTime;
+
+      /**
+       * Maximum last modified date of resource (inclusive).
+       */
+      @query("maxLastModifiedDateTime")
+      maxLastModifiedDateTime?: utcDateTime;
+
+      /**
+       * Maximum number of items needed (inclusive).
+       * Minimum = 10, Maximum = 1000, Default value = 50.
+       */
+      @maxValue(1000)
+      @minValue(10)
+      @query("maxPageSize")
+      maxPageSize?: int32 = 50;
+
+      /**
+       * Skip token for getting next set of results.
+       */
+      @query("skipToken")
+      skipToken?: string;
+    }>
+  >;
+
+  /**
+   * Gets a specified Catalog resource.
+   */
+  get is Operations.ResourceRead<Catalog>;
+
+  /**
+   * Creates or updates a Catalog resource.
+   */
+  createOrUpdate is Operations.ResourceCreateOrUpdate<Catalog>;
+
+  /**
+   * Deletes Catalog for given Catalog id.
+   */
+  delete is Operations.ResourceDelete<Catalog>;
+}
+
+interface CollectionsOperations {
+  /**
+   * Returns a paginated list of Collection resources.
+   */
+  list is Operations.ResourceList<
+    Collection,
+    QueryParametersTrait<{
+      /**
+       * Names of the resource.
+       */
+      #suppress "@azure-tools/typespec-azure-core/no-query-explode" ""
+      @query(#{ name: "names", explode: true })
+      names?: string[];
```

Comment:
Consider pulling out common query params into its own model instead of repeating them.